Marlin/src/gcode/control/M17_M18_M84.cpp

```diff
@@ -30,11 +30,21 @@
 #endif
 
 /**
- * M17: Enable power on all stepper motors
+ * M17: Enable stepper motors
  */
 void GcodeSuite::M17() {
-  LCD_MESSAGEPGM(MSG_NO_MOVE);
-  enable_all_steppers();
+  if (parser.seen("XYZE")) {
+    if (parser.seen('X')) enable_X();
+    if (parser.seen('Y')) enable_Y();
+    if (parser.seen('Z')) enable_Z();
+    #if HAS_E_STEPPER_ENABLE
+      if (parser.seen('E')) enable_e_steppers();
+    #endif
+  }
+  else {
+    LCD_MESSAGEPGM(MSG_NO_MOVE);
+    enable_all_steppers();
+  }
```

Comment:
In your research, did you discover why `M17` enables all steppers instead of being an extended command that takes axis parameters?

Reply:
Evidently it was originally added for compatibility with Replicator-G, so the spec is ad-hoc as with other M-codes. The description will need to be extended at https://reprap.org/wiki/G-code#M17:_Enable.2FPower_all_stepper_motors and https://github.com/MarlinFirmware/MarlinDocumentation/blob/master/_gcode/M017.md